src/components/feed/CreateFeed/CreateFeed.tsx

```diff
@@ -0,0 +1,507 @@
+};
+
+// UTILS
+
+export async function fetchAllChrisFiles(client: Client) {
+  const params = { limit: 100, offset: 0 };
+  let fileList = await client.getUploadedFiles(params);
+  const files = fileList.getItems() || [];
+
+  while (fileList.hasNextPage) {
+    try {
+      params.offset += params.limit;
+      fileList = await client.getUploadedFiles(params);
+      files.push(fileList.getItems());
+    } catch (e) {
+      console.error(e);
+    }
+  }
+  return files;
+}
+
+// INTERFACES
+
+export interface ChrisFile {
+  name: string,
+  path: string, // full path, including file name
+  id?: number, // only defined for files
+  children?: ChrisFile[],
+  collapsed?: boolean,
+}
+
+export interface LocalFile {
+  name: string,
+  blob: Blob
+}
+
+export type DataFile = ChrisFile | LocalFile;
+
+export interface CreateFeedData {
+  feedName: string,
+  feedDescription: string,
+  tags: Tag[],
+  chrisFiles: ChrisFile[],
+  localFiles: LocalFile[],
+}
+
+function getDefaultCreateFeedData(): CreateFeedData {
+  return {
+    feedName: '',
+    feedDescription: '',
+    tags: [],
+    chrisFiles: [],
+    localFiles: [],
+  }
+}
+
+interface CreateFeedProps {
+  authToken: string,
+  addFeed: (feed: IFeedItem) => void,
+}
+
+interface CreateFeedState {
+  wizardOpen: boolean,
+  saving: boolean,
+  step: number,
+  data: CreateFeedData
+}
+
+class CreateFeed extends React.Component<CreateFeedProps, CreateFeedState> {
+
+  client: Client = new Client(process.env.REACT_APP_CHRIS_UI_URL, { token: this.props.authToken });
+
+  constructor(props: CreateFeedProps) {
+    super(props);
+    this.state = {
+      wizardOpen: false,
+      saving: false,
+      step: 1,
+      data: getDefaultCreateFeedData()
+    }
+
+    this.toggleCreateWizard = this.toggleCreateWizard.bind(this);
+    this.handleStepChange = this.handleStepChange.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.getStepName = this.getStepName.bind(this);
+    this.handleFeedNameChange = this.handleFeedNameChange.bind(this);
+    this.handleFeedDescriptionChange = this.handleFeedDescriptionChange.bind(this);
+    this.handleTagsChange = this.handleTagsChange.bind(this);
+    this.handleChrisFileAdd = this.handleChrisFileAdd.bind(this);
+    this.handleChrisFileRemove = this.handleChrisFileRemove.bind(this);
+    this.handleLocalFilesAdd = this.handleLocalFilesAdd.bind(this);
+    this.handleLocalFileRemove = this.handleLocalFileRemove.bind(this);
+    this.createFeed = this.createFeed.bind(this);
+  }
+  /*
+    -------------- 
+    EVENT HANDLERS 
+    --------------
+  */
+
+  // WIZARD HANDLERS
+
+  resetState() {
+    this.setState({ 
+      data: getDefaultCreateFeedData(),
+      step: 1,
+      saving: false,
+    });
+  }
+
+  closeCreateWizard() {
+    this.setState({ wizardOpen: false });
+  }
+
+  toggleCreateWizard() {
+    if (this.state.wizardOpen) {
+      this.resetState();
+    }
+    this.setState({
+      wizardOpen: !this.state.wizardOpen
```

Comment:
Don't reference state in a setState method when using the object parameter since setState is an async call. Use the functional setState and reference prevState: 
setState((prevState, props) => { wizardOpen = !prevState.wizardOpen;});